src/System.Threading.Tasks.Extensions/src/System/Runtime/CompilerServices/AsyncBuilderAttribute.cs

```diff
@@ -5,14 +5,20 @@
 namespace System.Runtime.CompilerServices
 {
     /// <summary>
-    /// Indicates the async method builder that should be used by the C# compiler to build
-    /// the attributed type when used as the return type of an async method.
+    /// Indicates the async method builder that should be used by a language compiler to
+    /// build the attributed type when used as the return type of an async method.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface, Inherited = false, AllowMultiple = false)]
-    internal sealed class AsyncBuilderAttribute : Attribute
+    public sealed class AsyncBuilderAttribute : Attribute
```

Comment:
Just curious: will the language specification explicitly exclude enum and delegate types from being considered task-like (leaving it in the middle whether someone could come up with a meaningful builder for such types)? I assume that the detection is still name-based and not type identity based, so I could construct my own attribute with these targets and apply it to my task-like type?

One more question: assume that e.g. I want to experiment with making `IObservable<T>` task-like. Given that the attribute has to be applied to the task-like type, I won't be able to do this without modifying `mscorlib`. Could this be a concern to ship a library with support to make an existing type behave as task-like, without having to rev the library the type is defined in? (This feels very similar to the async extensions we shipped to make `Task` work with `await` using a `GetAwaiter` extension method.)